### PR TITLE
Update warning when users sign images by tag.

### PR DESCRIPTION
**Pull request by znewman01**
See #2047.

#### Summary
Signing by tag can lead to race conditions and possibly insecure behavior, signing images other than those intended. This PR adds a warning when people do that (later, we'll make it an error.)

#### Release Note
Added warning when users refer to images to sign by tag (`example.com/image:latest`). Instead, refer to images by digest (`example.com/image@sha256:abcdef`). This will be required in a future release.

#### Documentation

No change to web docs; changes to CLI docs included.

### CODE REVIEW

**Review by priyawadhwa — APPROVED** [review]

**Comment by znewman01** [comment]
See https://github.com/sigstore/cosign/pull/2314 for lint. Should be fixed once that's merged and I rebase

**Review by hectorj2f — DISMISSED** [review]
LGTM, I'm wondering if we should add a `NOTE` in the README, so any newbies are aware of this security recommendation.

**Comment by znewman01** [comment]
Newest change covers a lot of docs, and also adds some tests.

**Review by hectorj2f — DISMISSED** [review]

**Comment by cpanato** [comment]
will close/reopen to see if that retriggers the ci, otherwise we will need some rebase

**Review by cpanato — DISMISSED** [review]
thank you

**Comment by znewman01** [comment]
had to trigger tests 1 more time because the DCO check hung....fingers crossed!

**Comment by znewman01** [comment]
no substantive changes since hector last approved

**Review by cpanato — APPROVED** [review]
LGTM

thanks!

**Review by hectorj2f — APPROVED** [review]
Lgtm, thanks 💪

**Pull request merged**
